packages/chrome/src/elements/sheet/Sheet.spec.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useState, forwardRef } from 'react';
+import userEvent from '@testing-library/user-event';
+import { render } from 'garden-test-utils';
+
+import { Sheet, ISheetProps } from './Sheet';
+
+describe('Sheet', () => {
+  const Example = forwardRef<HTMLElement, ISheetProps>((props: ISheetProps, ref) => {
+    const [isOpen, setIsOpen] = useState(props.isOpen);
+    const buttonRef = useRef<HTMLButtonElement>(null);
+
+    return (
+      <>
+        <button ref={buttonRef} onClick={() => setIsOpen(!isOpen)}>
+          Toggle Sheet
+        </button>
+        <Sheet ref={ref} isOpen={isOpen} {...props}>
+          <Sheet.Header>
+            <Sheet.Title>title</Sheet.Title>
+            <Sheet.Description>description</Sheet.Description>
+          </Sheet.Header>
+          <Sheet.Body>body</Sheet.Body>
+          <Sheet.Footer>
+            <Sheet.FooterItem>
+              <button>button1</button>
+            </Sheet.FooterItem>
+            <Sheet.FooterItem>
+              <button>button2</button>
+            </Sheet.FooterItem>
+          </Sheet.Footer>
+          <Sheet.Close />
+        </Sheet>
+      </>
+    );
+  });
+
+  Example.displayName = 'Example';
+
+  it('passes ref to underlying DOM element', () => {
+    const ref = React.createRef<HTMLDivElement>();
+    const { getByRole, getByText } = render(<Example ref={ref} />);
+
+    userEvent.click(getByText('Toggle Sheet'));
+
+    expect(getByRole('complementary')).toBe(ref.current);
+  });
+
+  it('is only mounted when isOpen is true', () => {
+    const ref = React.createRef<HTMLDivElement>();
+    const { getByRole, getByText } = render(<Example ref={ref} />);
+
+    expect(() => getByRole('complementary')).toThrow();
+
+    userEvent.click(getByText('Toggle Sheet'));
+
+    expect(getByRole('complementary')).toBe(ref.current);
+  });
+
+  it('does not animate a transition when isAnimated is false', () => {
+    const { getByRole, getByText } = render(<Example isAnimated={false} />);
+
+    userEvent.click(getByText('Toggle Sheet'));
+
+    expect(getByRole('complementary')).not.toHaveAttribute('class', 'side-sheet-transition');
+  });
+
+  it('contains a11y bindings to label and describe the sheet', () => {
```

Comment:
Asserting against attributes (typically) side-steps `testing-library`'s [guiding principles](https://testing-library.com/docs/guiding-principles/) by asserting against implementation details. The user does not care if the title has a specific ID — they care if their browser is communicating that the title is mapped to the correct element.

The following would more align with `testing-library`'s guiding principles.

```jsx
it('has an accessible title and description', () => {
  render(<Example isOpen />);

  const sheet = screen.getByRole('complementary');

  // asserts the title is associated with the correct element
  expect(screen.getByLabelText('title')).toBe(sheet);

  // asserts the sheet has the correct description
  expect(sheet).toHaveDescription('description');
});
```

_Note 1: there is legacy Garden code that asserts against attributes directly that should be cleaned up_
_Note 2: may want to use modern `toHaveAccessibleDescription` with type casting_

Reply:
Thanks this is insightful! I've updated the test and need to review to see what I can do about other tests to best align with this.